### PR TITLE
feat(Elife): Add social sharing links

### PR DESCRIPTION
**Pull request by davidcmoulton**

### CODE REVIEW

File: src/themes/elife/index.ts [hunk 1]
```diff
@@ -39,4 +49,120 @@ ready((): void => {
       )
     )
   )
+
+  buildSocialSharers(getArticleTitle(), getArticleDoi())
 })
+
+const buildSocialSharers = (title: string, doi: string): void => {
+  const escapedTitle = encodeURIComponent(title)
+  const escapedDoi = encodeURI(`https://doi.org/${doi}`)
```
Comment (davidcmoulton) [inline src/themes/elife/index.ts:58]: `s/escaped/encoded/`

File: src/themes/elife/index.ts [hunk 2]
```diff
@@ -39,4 +49,126 @@ ready((): void => {
...
+
+  const attachPoint = first(':--Article :--title')
```
Comment (discodavey) [inline src/themes/elife/index.ts:60]: Looking on how Stencila do their coding they seem to put their clases together ie. ```:--Article:--title```

File: src/themes/elife/index.ts [hunk 3]
```diff
@@ -39,4 +49,126 @@ ready((): void => {
       )
     )
   )
+
+  buildSocialSharers(getArticleTitle(), getArticleDoi())
 })
+
+const buildSocialSharers = (title: string, doi: string): void => {
```
Comment (davidcmoulton) [inline src/themes/elife/index.ts:56]: Needs to be in its own module.

File: src/themes/elife/index.ts (hunk 2 above)
Comment (davidcmoulton) [reply src/themes/elife/index.ts:60]: Different use case: I want a `:--title` descended from an `:--Article`.

File: src/themes/elife/social-sharers.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+): void => {
+  const link = create('a', {
+    href: url,
+    class: 'social-sharer-link',
+    'aria-label': description,
+    target: '_blank',
+    rel: 'noopener noreferrer',
+  })
+  // Using DOM method innerHTML, not a Thema util method, as a way to get inline SVG text incorporated as an Element
+  link.innerHTML = imageMarkup
```
Comment (davidcmoulton) [inline src/themes/elife/social-sharers.ts:54]: @alex-ketch The requirement here is to be able to use text strings describing SVG images (e.g. [line 15](https://github.com/stencila/thema/pull/154/files#diff-9d2f0a0b8f0331f569f956da45255b4dR15)) to insert SVG elements into the DOM. I couldn't find a `util` method which would do that, hence addressing `innerHTML` directly here. Wondering if there's a more Thema-friendly way to do it?

File: src/themes/elife/index.ts [hunk 5]
```diff
@@ -22,6 +23,16 @@ const getArticleId = (): string => {
...
 }
 
+const getArticleDoi = (): string => {
+  const selector =
+    ':--identifier meta[content="https://registry.identifiers.org/registry/doi"] ~ [itemprop="value"]'
+  return first(selector)?.innerHTML ?? ''
+}
+
+const getArticleTitle = (): string => {
+  return first(':--title')?.innerHTML ?? ''
```
Comment (davidcmoulton) [inline src/themes/elife/index.ts:33]: `innerHTML` can include line breaks, which get encoded into the title string sent to social media. Thema's `text` util might be a better choice here.

Comment (davidcmoulton) [reply src/themes/elife/index.ts]: Addressed in 483bcbf.

File: src/themes/elife/social-sharers.ts (hunk 4 above)
Comment (alex-ketch) [reply src/themes/elife/social-sharers.ts:54]: @davidcmoulton you're right, seems like there currently isn't a utility for doing what you need.
Would you mind logging an issue please?

Comment (nokome) [reply src/themes/elife/social-sharers.ts:54]: The believe that the `create` function already does what you need:

```js
  const link = create('a', {
    href: url,
    class: 'social-sharer-link',
    'aria-label': description,
    target: '_blank',
    rel: 'noopener noreferrer',
  }, create(imageMarkup))
```

I created a PR to document and test that: https://github.com/stencila/thema/pull/155. Let me know if that isn't what you need.

Comment (davidcmoulton) [reply src/themes/elife/social-sharers.ts:54]: Thanks @nokome, just the job. Sorry, I missed that 😅

File: src/themes/elife/styles.css [hunk 6]
```diff
@@ -107,6 +108,36 @@ main {
...
 }
 
+:--Article .social-sharers {
+  height: 20px;
+  text-align: center;
+
+  /* Added 4px because height of social sharer link is 20px instead of 24px; */
+  margin-bottom: calc(var(--baseline-measure-rem) + 4px);
+
+  .social-sharer-link {
```
Comment (davidcmoulton) [inline src/themes/elife/styles.css:118]: Missing hover state and transition.

**Pull request merged**